common/beerocks/hostapd/include/configuration.h

```diff
@@ -192,11 +203,25 @@ class Configuration {
 };
 
 template <class func> void Configuration::for_all_ap_vaps(func f)
+{
+    auto f_with_iter = [&f](const std::string &vap, int iter) { f(vap); };
+
+    int dummy(0);
+    for_all_ap_vaps(f_with_iter, dummy, 10);
+}
+
+template <class func, class iter>
+void Configuration::for_all_ap_vaps(func f, iter current_iter, const iter end)
 {
     for_each(m_hostapd_config_vaps.begin(), m_hostapd_config_vaps.end(),
-             [this, &f](const std::pair<std::string, std::vector<std::string>> &vap) mutable {
+             [this, &f, &current_iter,
+              &end](const std::pair<std::string, std::vector<std::string>> &vap) {
                  if (get_vap_value(vap.first, "mode") == "ap") {
-                     f(vap.first);
+                     if (end == current_iter) {
+                         f(vap.first, end);
```

Comment:
Ah, actually this way indeed you can handle the remaining ones... But it's clunky because you need the condition inside, a separate function would be nicer.